zephyr/src/supp_events.c

```diff
@@ -27,3 +27,44 @@ int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int
 
 	return 0;
 }
+
+int send_wifi_state_event(const char *ifname, enum net_event_wpa_supp_cmd event, int status)
```

Comment:
Is `generate_wpa_supp_event` a more appropriate name for this function?

Reply:
`send_wifi_state_event` was used to keep it consistent with the other function which is handling wifi events. We can change this to `generate_wpa_supp_state_event` or `generate_supp_state_event`